src/Lean/Server/Rpc/Basic.lean

```diff
@@ -33,36 +40,98 @@ end Lean.Lsp
 
 namespace Lean.Server
 
-structure RpcObjectStore : Type where
-  /-- Objects that are being kept alive for the RPC client, together with their type names,
-  mapped to by their RPC reference.
+/--
+Marks values to be encoded as opaque references in RPC packets.
+Two `WithRpcRef`s with the same `id` will yield the same `RpcRef`.
+
+See also the docstring for `RpcEncodable`.
+-/
+structure WithRpcRef (α : Type u) where
+  private mk' ::
+    val : α
+    id  : USize
```

Comment:
```suggestion
    private id : USize
```
What do you think about making the projection private as well? It feels like other modules should not be able to observe this.